sonar-orchestrator/src/main/java/com/sonar/orchestrator/http/HttpCall.java

```diff
@@ -0,0 +1,216 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.orchestrator.http;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import okhttp3.Credentials;
+import okhttp3.FormBody;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.sonar.orchestrator.container.Server.ADMIN_LOGIN;
+import static com.sonar.orchestrator.container.Server.ADMIN_PASSWORD;
+import static java.lang.String.format;
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+public class HttpCall {
+
+  private static final String DEFAULT_USER_AGENT = "Orchestrator";
+
+  private final OkHttpClient okClient;
+  private final HttpUrl baseUrl;
+  private HttpMethod method = HttpMethod.GET;
+  private final Map<String, String> parameters = new LinkedHashMap<>();
+  private final Map<String, String> headers = new LinkedHashMap<>();
+  private Long timeoutMs = null;
+
+  HttpCall(OkHttpClient okClient, HttpUrl baseUrl) {
+    this.okClient = okClient;
+    this.baseUrl = baseUrl;
+    this.headers.put("User-Agent", DEFAULT_USER_AGENT);
+  }
+
+  public HttpUrl getBaseUrl() {
+    return baseUrl;
+  }
+
+  public HttpCall setMethod(HttpMethod m) {
+    this.method = m;
+    return this;
+  }
+
+  public HttpCall setCredentials(String login, @Nullable String password) {
+    headers.put("Authorization", Credentials.basic(login, password == null ? "" : password));
+    return this;
+  }
+
+  public HttpCall setAdminCredentials() {
+    return setCredentials(ADMIN_LOGIN, ADMIN_PASSWORD);
+  }
+
+  public HttpCall setAuthenticationToken(String token) {
+    return setCredentials(token, null);
+  }
+
+  public HttpCall setParam(String key, @Nullable String value) {
+    parameters.put(key, value);
+    return this;
+  }
+
+  public HttpCall setParams(String key1, @Nullable String value1, String... otherKeysAndNames) {
+    checkArgument(otherKeysAndNames.length % 2 == 0, "Expecting even number of arguments: %s", Arrays.toString(otherKeysAndNames));
+    parameters.put(key1, value1);
+    for (int i = 0; i < otherKeysAndNames.length; i++) {
+      parameters.put(otherKeysAndNames[i], otherKeysAndNames[i + 1]);
+      i++;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)